packages/legend-application-studio/src/stores/editor/editor-state/element-editor-state/service/ServiceRegistrationState.ts

```diff
@@ -100,7 +101,7 @@ export class ServiceRegistrationState {
 
   constructor(
     editorStore: EditorStore,
-    service: Service,
+    service: Service | undefined,
```

Comment:
why is service undefined here? cus i saw you use guaranteeNonNullable later

Reply:
I put undefined because BulkServiceRegistrationState extends ServiceRegistrationState and I have to pass in a service for the super. But bulkServiceReservationState would have many services

I also added the guarenteeNonNullable to be extra safe for the existing registration flow

Reply:
do you think it might better if you just pass in the first service from bulkServiceReservationState so that we could make less changes